OSDOCS-9991: adds metrics for (B)ANP (#76667)
openshift/openshift-docs: modules/nw-anp-banp-metrics.adoc

// Module included in the following assemblies:
//
// * list of assemblies where this module is included:
// networking/openshift_network_security/adminnetworkpolicy

:_mod-docs-content-type: CONCEPT
[id="anp-banp-metrics_{context}"]
= AdminNetworkPolicy and BaselineAdminNetworkPolicy metrics

The following table describes 12 metrics that can support monitoring your ANP and BANP.

[cols="1,1,1"]
|===
| Name |labels/action |Explanation

|`ovnkube_controller_admin_network_policies`
|Not applicable
|The total number of `AdminNetworkPolicy` in the cluster.

|`ovnkube_controller_baseline_admin_network_policies`
|Not applicable
|The total number of `BaselineAdminNetworkPolicy` in the cluster. Values should be 0 or 1.

|`ovnkube_controller_admin_network_policies_rules`
|*`direction`: specifies either ingress or egress.

> * `action`: specifies either `pass`, `allow`, or `deny`.

Reviewer: Could we capitalize actions - Pass, Allow and Deny?

|The total number of rules across all `AdminNetworkPolicy` in the cluster

|`ovnkube_controller_baseline_admin_network_policies_rules`
|*`direction`: specifies either ingress or egress.

> * `action`: specifies either `pass or `deny`.

Reviewer: Same as above, it is kind of reinforcement for the users to always capitalize the actions.

|The total number of rules across all `BaselineAdminNetworkPolicy` in the cluster

|`ovnkube_controller_admin_network_policies_db_objects`
|`tablename`

> |The total number of OVN Northbound database (nbdb) objects in the cluster.

Reviewer: We have two objects - ACLs and address sets

Reviewer: Table name is ACL or Address_Set

|`ovnkube_controller_baseline_admin_network_policies_db_objects`
|`tablename`

> |The total number of OVN Northbound database (nbdb) objects in the cluster.

Reviewer: Same we have two DB objects - ACLs and address sets

Reviewer: Table name is ACL or Address_Set

// Module included in the following assemblies:
//
// * list of assemblies where this module is included:
// networking/network_security/AdminNetworkPolicy/nw-ovn-k-anp-banp-metric.adoc

:_mod-docs-content-type: CONCEPT
[id="anp-banp-metrics_{context}"]

> = Metrics

Reviewer 2: Perhaps this could be a bit more descriptive -  maybe "AdminNetworkPolicy metrics" or something ?

[cols="1,1,1"]
|===

> | Name | Labels |Explanation

Reviewer 2: We usually see "Description" as the heading name for columns like this, just wondering if that would be appropriate here too?

|`ovnkube_controller_admin_network_policies`
|Not applicable

> |The total number of `AdminNetworkPolicy` in the cluster.

Reviewer 2: Can you qualify this? Would it be:

```suggestion
|The total number of `AdminNetworkPolicy` resources in the cluster.
```

Same for below

|`ovnkube_controller_baseline_admin_network_policies`
|Not applicable

> |The total number of `BaselineAdminNetworkPolicy` in the cluster. Values should be 0 or 1.

Reviewer 2: Is it really multiple values, or singular "the value"?

Author: yeah just value

|`ovnkube_controller_admin_network_policies_rules`

> |*`direction`: specifies either `Ingress` or `Egress`.

Reviewer 2: Are these asterisks supposed to be rendering as bullets? If so, you'd need to add an "a" next to the column number above that you want to render AsciiDoc

> |*`direction`: specifies either `Ingress` or `Egress`.

Reviewer 2: ```suggestion
|* `direction`: specifies either `Ingress` or `Egress`.
```

// Module included in the following assemblies:
//
// * list of assemblies where this module is included:
// networking/network_security/AdminNetworkPolicy/nw-ovn-k-anp-banp-metric.adoc

:_mod-docs-content-type: CONCEPT
[id="anp-banp-metrics_{context}"]

> = Metrics for AdminNetworkPolicy

Reviewer 3: just say Metrics

you say ANP and BANP everywhere :P even if its not needed but then added ANP alone here while table has BANP as well

Perhaps just make it "Metrics" since your heading is already monitoring ...?

Author: Metrics is fine, but I've been eliminating ANP and BANP :P.

> The following table describes 12 metrics that can support monitoring your ANP and BANP.

Reviewer 3: remove this line, redundant with the intro para

Author: agree

[cols="1,1,1"]
|===
| Name | Labels |Explanation

|`ovnkube_controller_admin_network_policies`
|Not applicable
|The total number of `AdminNetworkPolicy` in the cluster.

|`ovnkube_controller_baseline_admin_network_policies`
|Not applicable
|The total number of `BaselineAdminNetworkPolicy` in the cluster. Values should be 0 or 1.

|`ovnkube_controller_admin_network_policies_rules`

> |*`direction`: specifies either ingress or egress.

Reviewer 3: ```suggestion
|*`direction`: specifies either `Ingress` or `Egress`.
```

* `action`: specifies either `Pass`, `Allow`, or `Deny`.
|The total number of rules across all ANP policies in the cluster grouped by `direction` and `action`.

|`ovnkube_controller_baseline_admin_network_policies_rules`

> |*`direction`: specifies either ingress or egress.

Reviewer 3: ```suggestion
|*`direction`: specifies either `Ingress` or `Egress`.
```

* `action`: specifies either `Allow` or `Deny`.
|The total number of rules across all BANP policies in the cluster grouped by `direction` and `action`.

|`ovnkube_controller_admin_network_policies_db_objects`
|`tablename`

> |The total number of OVN Northbound database (nbdb) objects in the cluster which can be either `ACL` or `Address_Set`

Reviewer 3: ```suggestion
|The total number of OVN Northbound database (nbdb) objects that are created by all the admin network policies in the cluster.
```
 which can be either `ACL` or `Address_Set` are the labels needs to be the value of "table_name"

Author: I think I fixed this, but I'm not quite sure I fully got what you wanted. Please double check.

> |The total number of OVN Northbound database (nbdb) objects that are created by all the ANP in the cluster.

Reviewer 3: ```suggestion
|The total number of OVN Northbound database (nbdb) objects that are created by all the ANP in the cluster grouped by table_name.
```

|`ovnkube_controller_baseline_admin_network_policies_db_objects`
|`tablename`

> |The total number of OVN Northbound database (nbdb) objects in the cluster which can be either `ACL` or `Address_Set`

Reviewer 3: same as above comment

> |===

Reviewer 3: have all these `names` been double verified on a live cluster?

> |The total number of OVN Northbound database (nbdb) objects that are created by all the BANP in the cluster.

Reviewer 3: ```suggestion
|The total number of OVN Northbound database (nbdb) objects that are created by all the BANP in the cluster grouped by table_name.
```
keep it consistent what with you did for direction/action
